dpctl/_sycl_platform.pyx

```diff
@@ -0,0 +1,289 @@
+    DPCTLPlatform_GetBackend,
+    DPCTLPlatform_GetName,
+    DPCTLPlatform_GetPlatforms,
+    DPCTLPlatform_GetVendor,
+    DPCTLPlatform_GetVersion,
+    DPCTLPlatformMgr_PrintInfo,
+    DPCTLPlatformVector_Delete,
+    DPCTLPlatformVector_GetAt,
+    DPCTLPlatformVector_Size,
+    DPCTLPlatformVectorRef,
+    DPCTLSyclBackendType,
+    DPCTLSyclDeviceSelectorRef,
+    DPCTLSyclPlatformRef,
+)
+from . import backend_type
+
+__all__ = [
+    "get_platforms",
+    "lsplatform",
+    "SyclPlatform",
+]
+
+cdef class _SyclPlatform:
+    """ A helper metaclass to abstract a cl::sycl::platform instance.
+    """
+
+    def __dealloc__(self):
+        DPCTLPlatform_Delete(self._platform_ref)
+        DPCTLCString_Delete(self._name)
+        DPCTLCString_Delete(self._vendor)
+        DPCTLCString_Delete(self._version)
+
+
+cdef class SyclPlatform(_SyclPlatform):
+    """ Python equivalent for cl::sycl::platform class.
+    """
+    @staticmethod
+    cdef void _init_helper(SyclPlatform platform, DPCTLSyclPlatformRef PRef):
+        platform._platform_ref = PRef
+        platform._name = DPCTLPlatform_GetName(PRef)
+        platform._version = DPCTLPlatform_GetVersion(PRef)
+        platform._vendor = DPCTLPlatform_GetVendor(PRef)
+
+    @staticmethod
+    cdef SyclPlatform _create(DPCTLSyclPlatformRef pref):
+        cdef SyclPlatform p = <SyclPlatform>_SyclPlatform.__new__(_SyclPlatform)
+        # Initialize the attributes of the SyclPlatform object
+        SyclPlatform._init_helper(p, pref)
+        return SyclPlatform(p)
+
+    cdef int _init_from__SyclPlatform(self, _SyclPlatform other):
+        self._platform_ref = DPCTLPlatform_Copy(other._platform_ref)
+        if (self._platform_ref is NULL):
+            return -1
+        self._name = DPCTLPlatform_GetName(self._platform_ref)
+        self._version = DPCTLPlatform_GetVersion(self._platform_ref)
+        self._vendor = DPCTLPlatform_GetVendor(self._platform_ref)
+
+    cdef int _init_from_selector(self, DPCTLSyclDeviceSelectorRef DSRef):
+        # Initialize the SyclPlatform from a DPCTLSyclDeviceSelectorRef
+        cdef DPCTLSyclPlatformRef PRef = DPCTLPlatform_CreateFromSelector(DSRef)
+        if PRef is NULL:
+            return -1
+        else:
+            SyclPlatform._init_helper(self, PRef)
+            return 0
+
+    cdef DPCTLSyclPlatformRef get_platform_ref (self):
+        """ Returns the DPCTLSyclPlatformRef pointer for this class.
+        """
+        return self._platform_ref
+
+    @property
+    def __name__(self):
+        return "SyclPlatform"
+
+    def __repr__(self):
+        return ("<dpctl." + self.__name__ + " [" +
+                self.name + ", " + self.vendor +", " +
+                " " + self.version + "] at {}>".format(hex(id(self))) )
+
+    def __cinit__(self, arg=None):
+        if type(arg) is unicode:
+            string = bytes(<unicode>arg, "utf-8")
+            filter_c_str = string
+            DSRef = DPCTLFilterSelector_Create(filter_c_str)
```

Comment:
This branch and the next one share code, that can be factored out into a separate `_init_from_cstring` method

Reply:
done.